### ISSUE TITLE
[REVIEW]: treex: a Python package for manipulating rooted trees

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1351]
**Submitting author:** @RomainAzais  (<a href="http://orcid.org/0000-0002-5234-1822">Romain Azais</a>)
**Repository:** <a href="https://gitlab.inria.fr/azais/treex" target ="_blank">https://gitlab.inria.fr/azais/treex</a>
**Version:** 1.2.0
**Editor:** @VivianePons
**Reviewers:** @VivianePons, @dvberkel
**Archive:** <a href="https://doi.org/10.5281/zenodo.3254416" target="_blank">10.5281/zenodo.3254416</a>

## Status

[![status](http://joss.theoj.org/papers/0800c3bd9442b5e21b607776d7df666a/status.svg)](http://joss.theoj.org/papers/0800c3bd9442b5e21b607776d7df666a)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/0800c3bd9442b5e21b607776d7df666a"><img src="http://joss.theoj.org/papers/0800c3bd9442b5e21b607776d7df666a/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/0800c3bd9442b5e21b607776d7df666a/status.svg)](http://joss.theoj.org/papers/0800c3bd9442b5e21b607776d7df666a)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://gitlab.inria.fr/azais/treex" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@dvberkel  & @VivianePons, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @VivianePons know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://gitlab.inria.fr/azais/treex">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 1.2.0
- [x] **Authorship:** Has the submitting author (@azais) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [X] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [X] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [X] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://gitlab.inria.fr/azais/treex">repository url</a>?
- [X] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [X] **Version:** 1.2.0
- [X] **Authorship:** Has the submitting author (@azais) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [X] **Installation:** Does installation proceed as outlined in the documentation?
- [X] **Functionality:** Have the functional claims of the software been confirmed?
- [X] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [X] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [X] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [X] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [X] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [X] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @dvalters, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01351/joss.01351/10.21105.joss.01351.pdf)

**Comment by VivianePons** [comment]
Hi @Azais 

I am trying to install the software but I am running into issues. I have a miniconda on my machine but I am quite new with it and I can't find how to upgrade the different lib. I created an environment for treex but when I try the install command, I run into:

```
Fetching package metadata .............
Solving package specifications: 

PackageNotFoundError: Packages missing in current channels:
            
  - treex -> libedit >=3.1.20181209,<3.2.0a0
  - treex -> openssl >=1.1.1a,<1.1.2a -> libgcc-ng >=7.3.0
  - treex -> sqlite >=3.26.0,<4.0a0
  - treex -> xz >=5.2.4,<6.0a0

We have searched for the packages in the following channels:
            
  - https://conda.anaconda.org/mosaic/linux-64
  - https://conda.anaconda.org/mosaic/noarch
  - https://conda.anaconda.org/conda-forge/linux-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.continuum.io/pkgs/free/linux-64
  - https://repo.continuum.io/pkgs/free/noarch
  - https://repo.continuum.io/pkgs/r/linux-64
  - https://repo.continuum.io/pkgs/r/noarch
  - https://repo.continuum.io/pkgs/pro/linux-64
  - https://repo.continuum.io/pkgs/pro/noarch
```

Maybe I am just missing the right channels. As I said, I usually don't use conda that much. I only had a miniconda install for another project and it might be very minimal. Thanks for your help.

**Comment by VivianePons** [comment]
Note: when I try installing libedit for example, which seems to be required, with this command 

``` conda install -c conda-forge libedit```

it only installs version 3.1.20170329 , which is not good enough

**Comment by RomainAzais** [comment]
Hello @VivianePons
Could you try to update conda with `conda update conda`?
If it doesn't work, could you comment with the output of `conda info`?
Thanks.

**Comment by VivianePons** [comment]
Thanks, it worked!

**Comment by VivianePons** [comment]
I have started reviewing. I am not finished but I have some remarks.

First, it would be good to have the link to the documentation and the wiki inside the main README. Actually, I think the install should be described directly in the repo and not depend on the wiki (even though, it's good to also have the wiki)

And now, a more specific question. If I create a tree t and add a subtree s, I notice that the id of s is kept.

```
In [11]: t = Tree()                                                             

In [12]: s = Tree()                                                             

In [13]: print(t)                                                               
140108254638880


In [14]: print(s)                                                               
140108194820336


In [15]: t.add_subtree(s)                                                       

In [16]: print(t)                                                               
140108254638880
	140108194820336
```

But, if I add s though the `add_subtree_to_id` method, then, the new subtree in t is not s!

```
In [5]: t = Tree()                                                              

In [6]: s = Tree()                                                              

In [7]: print(t)                                                                
140108254475768


In [8]: print(s)                                                                
140108280308736


In [9]: t.add_subtree_to_id(s, t.my_id)                                         

In [10]: print(t)                                                               
140108254475768
	140108254642128
```

Why is that? Is this a bug of a feature? If it's a feature, it see it nowhere explained on the documentation of the method.

**Comment by RomainAzais** [comment]
Hello Viviane,
Thank you for this first feedback.
- About add_subtree VS add_subtree_to_id: when constructing a tree, we want the ids to be unique. But, we also thought that it should be a good idea to keep the option of replicating the id. I agree that our code is not clear on that point. We can change this (only one method called add_subtree, with an optional variable for copying or not the id).
- OK for the readme file.
How are we expected to process? We can do a new release today for instance.
Romain

**Comment by VivianePons** [comment]
Yes, I understand the necessity of the two options. My comment was mostly reflecting on the fact that this is not reflected clearly in the documentation. 

You can indeed release a new version.

**Comment by RomainAzais** [comment]
Hello Viviane,

Sorry for my late reply. We released the new version about one week ago, but we got some issues with virtual machines for continuous integration, especially for MacOSX.

In the new version, the two methods add_subtree and add_subtree_to_id (where only add_subtree_to_id copies the subtree to be added) are still present, but the documentation has been clarified on this point:
- https://azais.gitlabpages.inria.fr/treex/_dvlpt/treex.tree.html#treex.tree.Tree.add_subtree
- https://azais.gitlabpages.inria.fr/treex/_dvlpt/treex.tree.html#treex.tree.Tree.add_subtree_to_id
In addition, the variable node_id is now optional in add_subtree_to_id. If None, the subtree is copied then added to the root of the main tree.

The installation procedure is now described in the ReadMe file. We also added the URL of the doc and the wiki.

All the best

**Comment by VivianePons** [comment]
Thanks, sorry I was also a bit busy here. I will complete my review next week.

@dvalters did you have time to start?

**Comment by VivianePons** [comment]
@whedon set 1.1.4 as version

**Comment by whedon** [comment]
OK. 1.1.4 is the version.

**Comment by VivianePons** [comment]
I have a question for @arfon : the software license is CeCILL-C whereas the OSI accepted one is CeCILL-2.1. Is there a significant difference there, is this an issue at all?

**Comment by VivianePons** [comment]
@whedon check doi

**Comment by VivianePons** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1137/1.9781611975499.4 is OK
- 10.3389/fpls.2012.00076 is OK

MISSING DOIs

- https://doi.org/10.1109/tcbb.2009.29 may be missing for title: Quantifying the degree of self-nestedness of trees: application to the structural analysis of plants
- https://doi.org/10.30757/alea.v16-21 may be missing for title: Inference for conditioned Galton-Watson trees from their Harris path
- https://doi.org/10.1016/0304-4149(94)90101-5 may be missing for title: Combinatorial stochastic processes

INVALID DOIs

- None
```

**Comment by VivianePons** [comment]
@RomainAzais could you please add the missing DOI?

**Comment by VivianePons** [comment]
@whedon remove @dvalters  as reviewer

**Comment by whedon** [comment]
OK, @dvalters  is no longer a reviewer

**Comment by VivianePons** [comment]
@whedon add @dvberkel as reviewer

**Comment by whedon** [comment]
OK, @dvberkel is now a reviewer

**Comment by RomainAzais** [comment]
> @RomainAzais could you please add the missing DOI?
Done!

Note that the DOI found for "Combinatorial stochastic processes" in:
> ``` 
> MISSING DOIs
> 
> - https://doi.org/10.1109/tcbb.2009.29 may be missing for title: Quantifying the degree of self-nestedness of trees: application to the structural analysis of plants
> - https://doi.org/10.30757/alea.v16-21 may be missing for title: Inference for conditioned Galton-Watson trees from their Harris path
> - https://doi.org/10.1016/0304-4149(94)90101-5 may be missing for title: Combinatorial stochastic processes
> ```
was not the good one. I added the right DOI.

**Comment by dvberkel** [comment]
@RomainAzais I was looking at `CONTRIBUTING.rst` and it mentions and links to GitHub instead of GitLab.

It might be the case that you hosted it on GitHub earlier and then moved to GitLab? Although it is a minor inconvenience, I would love to see the `CONTRIBUTING.rst` reflect the current situation.

**Comment by VivianePons** [comment]
> > @RomainAzais could you please add the missing DOI?
> 
> Done!
> 
> Note that the DOI found for "Combinatorial stochastic processes" in
> 
> > ```
> > MISSING DOIs
> > 
> > - https://doi.org/10.1109/tcbb.2009.29 may be missing for title: Quantifying the degree of self-nestedness of trees: application to the structural analysis of plants
> > - https://doi.org/10.30757/alea.v16-21 may be missing for title: Inference for conditioned Galton-Watson trees from their Harris path
> > - https://doi.org/10.1016/0304-4149(94)90101-5 may be missing for title: Combinatorial stochastic processes
> > ```
> 
> was not the good one. I added the right DOI.

Perfect, thank you :)

**Comment by VivianePons** [comment]
@arfon : have you seen my question about the license? I believe CeCILL-C is ok instead of CeCILL-2.1 but I am no expert...

**Comment by arfon** [comment]
> @arfon : have you seen my question about the license? I believe CeCILL-C is ok instead of CeCILL-2.1 but I am no expert...

I don't know about any differences sorry. If CeCILL-2.1 is the only OSI-approved one then we would need the authors to switch to that license before we can accept this submission.

**Comment by RomainAzais** [comment]
> @RomainAzais I was looking at `CONTRIBUTING.rst` and it mentions and links to GitHub instead of GitLab.
> 
> It might be the case that you hosted it on GitHub earlier and then moved to GitLab? Although it is a minor inconvenience, I would love to see the `CONTRIBUTING.rst` reflect the current situation.

You're right.
Done!

**Comment by RomainAzais** [comment]
> > @arfon : have you seen my question about the license? I believe CeCILL-C is ok instead of CeCILL-2.1 but I am no expert...
> 
> I don't know about any differences sorry. If CeCILL-2.1 is the only OSI-approved one then we would need the authors to switch to that license before we can accept this submission.

@VivianePons So what should we do with the license?

**Comment by VivianePons** [comment]
Is it possible to change it to CeCILL-2.1? From what I understood, this is mostly due to translation issue. CeCILL-C was written to comply with the French law. CeCILL-2.1 is now working both in English and French and complies well with the French law and international norms (such as OSI).

**Comment by RomainAzais** [comment]
> Is it possible to change it to CeCILL-2.1? From what I understood, this is mostly due to translation issue. CeCILL-C was written to comply with the French law. CeCILL-2.1 is now working both in English and French and complies well with the French law and international norms (such as OSI).

We switched to LGPL that is (more or less) equivalent to CeCILL-C but OSI approved: https://opensource.org/licenses/LGPL-3.0

**Comment by VivianePons** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1137/1.9781611975499.4 is OK
- 10.3389/fpls.2012.00076 is OK

MISSING DOIs

- https://doi.org/10.1109/tcbb.2009.29 may be missing for title: Quantifying the degree of self-nestedness of trees: application to the structural analysis of plants
- https://doi.org/10.30757/alea.v16-21 may be missing for title: Inference for conditioned Galton-Watson trees from their Harris path
- https://doi.org/10.1016/0304-4149(94)90101-5 may be missing for title: Combinatorial stochastic processes

INVALID DOIs

- None
```

**Comment by VivianePons** [comment]
Perfect for the license!

It seems that there are still some missing DOI, can you check?

**Comment by RomainAzais** [comment]
> It seems that there are still some missing DOI, can you check?

The way that the DOIs are interpreted is quite unclear for me, but it should be OK now.
Please note that the DOI found for 'Combinatorial stochastic processes' is not the good one. So a new issue could be raised.

**Comment by VivianePons** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/TCBB.2009.29 is OK
- 10.1137/1.9781611975499.4 is OK
- 10.30757/ALEA.v16-21 is OK
- 10.1007/b11601500 is OK
- 10.3389/fpls.2012.00076 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by VivianePons** [comment]
Everything is good on my side, @dvberkel are you done with your review? I'm waiting for your green light to move forward with the process!

**Comment by dvberkel** [comment]
I am done with my review. I am giving my green light to move forward with the process! :+1:

**Comment by VivianePons** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01351/joss.01351/10.21105.joss.01351.pdf)

**Comment by VivianePons** [comment]
Perfect :) @RomainAzais could you make a zenodo archive and give me the reference? Please make it so that the metadata of the archive (title and authors) match the one of the paper. Thanks

**Comment by RomainAzais** [comment]
Hello!
Thanks for your positive reviews and comments.
I have few questions on the article proof:
- The link `Archive` in the left column (under Software) is not active. It should point to [https://anaconda.org/MOSAIC/treex].
- Could it be possible to add a link to the documentation ([https://azais.gitlabpages.inria.fr/treex]) at this location (left column, under Software)?
- Since the repository and the archive (and maybe the documentation) are available from the left column, can I remove them from the list of references?

**Comment by arfon** [comment]
Hi @RomainAzais, the `Archive` link will become active when you make an archive of the software as requested by @VivianePons in [this comment above](https://github.com/openjournals/joss-reviews/issues/1351#issuecomment-503943781). 

We ask that all authors make an archive of their software at the end of their review using a service such as figshare or Zenodo.

> Could it be possible to add a link to the documentation (https://azais.gitlabpages.inria.fr/treex) at this location (left column, under Software)?

I'm afraid we don't support customizing the JOSS papers in this way.

> Since the repository and the archive (and maybe the documentation) are available from the left column, can I remove them from the list of references?

Yes, that's fine.

**Comment by RomainAzais** [comment]
> Hi @RomainAzais, the `Archive` link will become active when you make an archive of the software as requested by @VivianePons in [this comment above](https://github.com/openjournals/joss-reviews/issues/1351#issuecomment-503943781).

OK, understood.

> I'm afraid we don't support customizing the JOSS papers in this way.

No worries.

> Yes, that's fine.

OK. I'll come back with the new version and the Zenodo archive.

**Comment by VivianePons** [comment]
Just to let you know that I'm still here. I set up the ooo because I'm not taking new papers but whenever you have the zenodo, I will finish it up here (and update the version number accordingly)

**Comment by RomainAzais** [comment]
I updated the paper and the bibtex file according to the comments above.
Here is the link of the Zenodo archive: https://zenodo.org/record/3254416 (the DOI is 10.5281/zenodo.3254416).
Let me know if something goes wrong.

**Comment by VivianePons** [comment]
@whedon set 10.5281/zenodo.3254416  as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3254416" target="_blank">10.5281/zenodo.3254416</a> is the archive.

**Comment by VivianePons** [comment]
@whedon set 1.2.0 as version

**Comment by whedon** [comment]
OK. 1.2.0 is the version.

**Comment by VivianePons** [comment]
Perfect :) :) Your paper is now officially accepted in JOSS. Thank you very much all of you for your work on this! I leave it to @openjournals/joss-eics

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/TCBB.2009.29 is OK
- 10.1137/1.9781611975499.4 is OK
- 10.30757/ALEA.v16-21 is OK
- 10.1007/b11601500 is OK
- 10.3389/fpls.2012.00076 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/792

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/792, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1143155139412484096) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/793
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01351](https://doi.org/10.21105/joss.01351)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@VivianePons, @dvberkel - many thanks for your reviews here and to @VivianePons for editing the submission too ✨ 

@RomainAzais - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01351/status.svg)](https://doi.org/10.21105/joss.01351)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01351">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01351/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01351/status.svg
   :target: https://doi.org/10.21105/joss.01351
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01351/status.svg)](https://doi.org/10.21105/joss.01351)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss

**Comment by RomainAzais** [comment]
🎊 🎊 🍸 
Thank you very much @arfon, @dvberkel, and @VivianePons for your work!
All the best,
Romain
